### PR TITLE
Mail template fix "reply to"

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue #37473 .

### Summary of Changes
Mail template should use replyto from config, when available

### Testing Instructions

Please follow #37473
Or add "Reply-To" in Global config, save (!!), and send test email.

### Actual result BEFORE applying this Pull Request
Reply-To is missing

### Expected result AFTER applying this Pull Request
Reply-To is there

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 0e2aa376acdbd9625cfbee6ef42a51477c4a0e2d<br /><br />Tested successfully in Joomla 4.2.28-dev of 20 February in Wampserver 3.2.9 using PHP 8.1.10<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39893">issues.joomla.org/tracker/joomla-cms/39893</a>.</sub>

File: libraries/src/Mail/MailTemplate.php [hunk 1]
```diff
@@ -196,8 +196,10 @@ public function send()
         }
 
         /** @var Registry $params */
-        $params = $mail->params;
-        $app    = Factory::getApplication();
+        $params      = $mail->params;
+        $app         = Factory::getApplication();
+        $replyto     = $app->get('replyto', '');
+        $replytoname = $app->get('replytoname', '');
```
Comment (joomdonation) [inline libraries/src/Mail/MailTemplate.php]: Shouldn't we follow camel case naming convention for variables here? So use `$replyTo` and `$replyToName` ?

File: libraries/src/Mail/MailTemplate.php [hunk 2]
```diff
@@ -278,6 +283,8 @@ public function send()
 
         if ($this->replyto) {
             $this->mailer->addReplyTo($this->replyto->mail, $this->replyto->name);
+        } elseif ($replyto) {
```
Comment (joomdonation) [inline libraries/src/Mail/MailTemplate.php]: Is `MailHelper::isEmailAddress` needed here?

File: libraries/src/Mail/MailTemplate.php (hunk 1 above)
Comment (Fedik) [reply libraries/src/Mail/MailTemplate.php]: I made as it was there, same with `$mailfrom` which more down

File: libraries/src/Mail/MailTemplate.php (hunk 2 above)
Comment (Fedik) [reply libraries/src/Mail/MailTemplate.php]: It already validated when saving in global config.

Comment (joomdonation) [reply libraries/src/Mail/MailTemplate.php]: Ah, Yes.

File: libraries/src/Mail/MailTemplate.php (hunk 1 above)
Comment (joomdonation) [reply libraries/src/Mail/MailTemplate.php]: I still think that for the **_new_** variables should follow the convention, not depend on existing variables. But never mind of no one else complains.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 0e2aa376acdbd9625cfbee6ef42a51477c4a0e2d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39893">issues.joomla.org/tracker/joomla-cms/39893</a>.</sub>

File: libraries/src/Mail/MailTemplate.php (hunk 1 above)
Comment (roland-d) [reply libraries/src/Mail/MailTemplate.php]: @Fedik I do agree with @joomdonation that for **new** variables we should follow the convention of camelCase. This way we do not introduce more work for later.

Comment (Fedik) [reply libraries/src/Mail/MailTemplate.php]: changed

**Comment by joomdonation** [comment]
@toivo or @ChristineWk Could one of you help re-testing this PR ? After that, it will be ready to be merged.

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 523b6a22074f2ff216c7cc8a894bc59bfa0c719e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39893">issues.joomla.org/tracker/joomla-cms/39893</a>.</sub>

**Comment by ChristineWk** [comment]
Screenshot 2023-02-23 110913

**Comment by joomdonation** [comment]
Thanks @ChristineWk. RTC because the last commit was just a simple rename variables and we now have one actual human tested it with successful test.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39893">issues.joomla.org/tracker/joomla-cms/39893</a>.</sub>

**Comment by roland-d** [comment]
Thank you

**Pull request merged**